Description: changes already applied to upstream for easing debianization
 Including upstream commits:
  * https://github.com/HaxeFoundation/haxe/commit/4ad44f4dc651af104adedb25c59e5701ed227b1d
  * https://github.com/HaxeFoundation/haxe/commit/5c99ce25977748922a2c7d3be16d6164c244b9e6
  * https://github.com/HaxeFoundation/haxe/commit/9ab0e444dcb9788abb3baf5a1e47321ec51d7207
Author: Andy Li <andy@onthewings.net>

--- haxe-3.2.0+dfsg.orig/Makefile
+++ haxe-3.2.0+dfsg/Makefile
@@ -10,9 +10,10 @@
 #
 .SUFFIXES : .ml .mli .cmo .cmi .cmx .mll .mly
 
-INSTALL_DIR=/usr
+INSTALL_DIR=$(DESTDIR)/usr
 INSTALL_BIN_DIR=$(INSTALL_DIR)/bin
 INSTALL_LIB_DIR=$(INSTALL_DIR)/lib/haxe
+INSTALL_STD_DIR=$(INSTALL_LIB_DIR)/std
 
 OUTPUT=haxe
 EXTENSION=
@@ -89,12 +90,12 @@ haxelib:
 tools: haxelib
 
 install:
-	-rm -f $(INSTALL_LIB_DIR)
-	-mkdir -p $(INSTALL_LIB_DIR)
-	rm -rf $(INSTALL_LIB_DIR)/std
-	cp -rf std $(INSTALL_LIB_DIR)/std
+	rm -rf $(INSTALL_LIB_DIR)
+	mkdir -p $(INSTALL_BIN_DIR)
+	mkdir -p $(INSTALL_LIB_DIR)/lib
+	rm -rf $(INSTALL_STD_DIR)
+	cp -rf std $(INSTALL_STD_DIR)
 	cp -rf extra $(INSTALL_LIB_DIR)
-	-mkdir -p $(INSTALL_LIB_DIR)/lib
 	rm -f $(INSTALL_BIN_DIR)/haxe
 	cp haxe $(INSTALL_LIB_DIR)
 	ln -s $(INSTALL_LIB_DIR)/haxe $(INSTALL_BIN_DIR)/haxe
@@ -175,7 +176,7 @@ lexer.$(MODULE_EXT): ast.$(MODULE_EXT)
 ast.$(MODULE_EXT):
 
 version.$(MODULE_EXT):
-	$(MAKE) -f Makefile.version_extra -s ADD_REVISION=$(ADD_REVISION) BRANCH=$(BRANCH) COMMIT_SHA=$(COMMIT_SHA) COMMIT_DATE=$(COMMIT_DATE) > version.ml
+	$(MAKE) -f Makefile.version_extra -s --no-print-directory ADD_REVISION=$(ADD_REVISION) BRANCH=$(BRANCH) COMMIT_SHA=$(COMMIT_SHA) COMMIT_DATE=$(COMMIT_DATE) > version.ml
 	$(COMPILER) $(CFLAGS) -c version.ml
 
 # Package
@@ -207,10 +208,12 @@ clean_libs:
 	make -C libs/objsize clean
 
 clean_haxe:
-	rm -f $(MODULES:=.obj) $(MODULES:=.o) $(MODULES:=.cmx) $(MODULES:=.cmi) $(MODULES:=.cmo) lexer.ml $(OUTPUT)
+	rm -f $(MODULES:=.obj) $(MODULES:=.o) $(MODULES:=.cmx) $(MODULES:=.cmi) $(MODULES:=.cmo) lexer.ml version.ml $(OUTPUT)
 
 clean_tools:
 	rm -f $(OUTPUT) haxelib
+	rm -f extra/haxelib_src/bin/haxelib.n
+	rm -f extra/haxelib_src/bin/haxelib
 
 # SUFFIXES
 
